File: src/main/java/ar/edu/um/tesoreria/rest/extern/model/InscripcionFacultad.java
/**
 * 
 */
package ar.edu.um.tesoreria.rest.extern.model;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.OffsetDateTime;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author daniel
 *
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class InscripcionFacultad implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 2430193945316706645L;

	private Integer facultadId;
	private BigDecimal personaId;
	private Integer documentoId;
	private Integer lectivoId;
	private Long inscripcionId;
	private OffsetDateTime fecha;
	private String chequera = "";
	private String matricula = "";
	private Long factura = 0L;
	private Integer curso = 0;
	private Integer planId;
	private Integer carreraId;
	private Integer geograficaId;
	private Byte asentado = 0;
	private Byte provisoria = 0;
	private Integer cohorte = 0;
	private Byte remota = 0;
	private Byte imprimir = 0;
	private Integer edad = 0;
	private String observaciones = "";
	private Integer offsetpago = 0;
	private Integer libre = 0;
	private Integer divisionId;
	private Byte debematricula = 0;

	public String getPersonaKey() {
		return this.personaId + "." + this.documentoId;
	}

	public String getCarreraKey() {
		return this.facultadId + "." + this.planId + "." + this.carreraId;
	}

	public String getPlanKey() {
		return this.facultadId + "." + this.planId;
	}

	public String getLegajoKey() {
		return this.facultadId + "." + this.personaId + "." + this.documentoId;
	}

}

File: src/main/java/um/tesoreria/rest/extern/model/kotlin/InscripcionFacultad.kt
package um.tesoreria.rest.extern.model.kotlin

import java.math.BigDecimal
import java.time.OffsetDateTime

data class InscripcionFacultad(

    var facultadId: Int? = null,
    var personaId: BigDecimal? = null,
    var documentoId: Int? = null,
    var lectivoId: Int? = null,
    var inscripcionId: Long? = null,
    var fecha: OffsetDateTime? = null,
    var chequera: String = "",
    var matricula: String = "",
    var factura: Long = 0L,
    var curso: Int = 0,
    var planId: Int? = null,
    var carreraId: Int? = null,
    var geograficaId: Int? = null,
    var asentado: Byte = 0,
    var provisoria: Byte = 0,
    var cohorte: Int = 0,
    var remota: Byte = 0,
    var imprimir: Byte = 0,
    var edad: Int = 0,
    var observaciones: String = "",
    var offsetpago: Int = 0,
    var libre: Int = 0,
    var divisionId: Int? = null,
    var debematricula: Byte = 0,

) {

    fun getPersonaKey(): String {
        return personaId.toString() + "." + documentoId
    }

    fun getCarreraKey(): String {
        return facultadId.toString() + "." + planId + "." + carreraId
    }

    fun getPlanKey(): String {
        return facultadId.toString() + "." + planId
    }

    fun getLegajoKey(): String {
        return facultadId.toString() + "." + personaId + "." + documentoId
    }

}
